src/overlays/actors/ovl_Bg_Spot09_Obj/z_bg_spot09_obj.c

```diff
@@ -21,25 +24,165 @@ const ActorInit Bg_Spot09_Obj_InitVars = {
     (ActorFunc)BgSpot09Obj_Update,
     (ActorFunc)BgSpot09Obj_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot09_Obj/func_808B1AE0.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot09_Obj/func_808B1BA0.s")
+static u32 D_808B1F90[] = { 0x00000000, 0x06005520, 0x0600283C, 0x06008458, 0x06007580 };
+
+typedef s32 (*ActorFuncRet)(struct Actor*, struct GlobalContext*);
+static ActorFuncRet D_808B1FA4[] = {
+    (ActorFuncRet)func_808B1BEC,
+    (ActorFuncRet)func_808B1AE0,
+    (ActorFuncRet)func_808B1BA0,
+};
+
+static InitChainEntry initChain1[] = {
+    ICHAIN_F32(unk_F4, 7200, ICHAIN_CONTINUE),
+    ICHAIN_F32(unk_F8, 3000, ICHAIN_CONTINUE),
+    ICHAIN_F32(unk_FC, 7200, ICHAIN_STOP),
+};
+
+static InitChainEntry initChain2[] = {
+    ICHAIN_F32(unk_F4, 7200, ICHAIN_CONTINUE),
+    ICHAIN_F32(unk_F8, 800, ICHAIN_CONTINUE),
+    ICHAIN_F32(unk_FC, 1500, ICHAIN_STOP),
+};
+
+static u32 dlists[] = {
+    0x06000100, 0x06003970, 0x06001120, 0x06007D40, 0x06006210,
+};
+
+extern UNK_TYPE D_06008010;
+
+static s32 func_808B1AE0(BgSpot09Obj* this, GlobalContext* globalCtx) {
+    s32 temp_v1;
+    s32 linkAge;
+    Actor* thisx = &this->dyna.actor;
+
+    if (gSaveContext.scene_setup_index >= 4) {
+        return thisx->params == 0;
+    }
+
+    temp_v1 = (gSaveContext.event_chk_inf[9] & 0xF) == 0xF;
+
+    if (LINK_IS_CHILD) {
+        linkAge = YEARS_CHILD;
+    } else {
+        linkAge = YEARS_ADULT;
+    }
+
+    if (linkAge == YEARS_ADULT) {
```

Comment:
As mentioned in discord the LINK_AGE_IN_YEARS macro should be used alongside YEARS_CHILD/YEARS_ADULT. The temp variable shouldn't be needed either. If it is then this is a special case and we will have to figure out an alternative pattern. But anyway here is the ideal pattern:
```suggestion
    if (LINK_AGE_IN_YEARS == YEARS_ADULT) {
```